bee-message/src/output/mod.rs

```diff
@@ -201,10 +204,71 @@ impl Output {
             Self::Nft(output) => Some(output.immutable_feature_blocks()),
         }
     }
+
+    /// Checks if a sufficient storage deposit was made for the given [`Output`].
+    pub fn check_sufficient_storage_deposit(&self, config: &ByteCostConfig) -> Result<(), Error> {
+        let required_output = self.byte_cost(config);
+
+        if self.amount() < required_output {
+            return Err(Error::InsufficientStorageDepositAmount {
+                amount: self.amount(),
+                required: required_output,
+            });
+        }
+
+        if let Some(return_condition) = self
+            .unlock_conditions()
+            .and_then(UnlockConditions::storage_deposit_return)
+        {
+            // We can't return more tokens than were originally contained in the output.
+            // `0` ≤ `Amount` - `Return Amount`
+            if return_condition.amount() > self.amount() {
+                return Err(Error::StorageDepositReturnExceedsOutputAmount {
+                    deposit: return_condition.amount(),
+                    amount: self.amount(),
+                });
+            }
+
+            let minimum_deposit = minimum_storage_deposit(config, return_condition.return_address());
+
+            // `Return Amount` must be ≥ than `Minimum Storage Deposit`
+            if return_condition.amount() < minimum_deposit {
+                return Err(Error::InsufficientStorageDepositReturnAmount {
+                    deposit: return_condition.amount(),
+                    required: minimum_deposit,
+                });
+            }
+
+            // Check if the storage deposit return was required in the first place.
+            // `Amount` - `Return Amount` ≤ `Required Storage Deposit of the Output`
+            if self.amount() - return_condition.amount() > required_output {
+                return Err(Error::UnnecessaryStorageDepositReturnCondition {
+                    remainder: self.amount() - return_condition.amount(),
```

Comment:
I don't like the name `remainder` here, because this will be the amount the receiver gets and remainder would rather be the amount in the `StorageDepositReturn`

Reply:
How about the `logical_amount` of an `Output`, instead of `remainder`?

Reply:
Would be fine for me